File: ddingmin/baekjoon/14658.kt
import java.io.BufferedWriter
import java.io.OutputStreamWriter

fun main() = with(System.`in`.bufferedReader()) {
    val sout = BufferedWriter(OutputStreamWriter(System.out))

    var line = readLine().split(" ").map { it.toInt() }
    var l = line[2]
    var k = line[3]

    var arr = Array(k) {
        readLine().split(" ").map { it.toInt() }
    }

    sout.appendLine(solve(l, k, arr))
    sout.close()
}

fun solve(l: Int, k: Int, arr: Array<List<Int>>): String {
    var count = 0

    for (star1 in arr) {
        for (star2 in arr) {
            var sx = minOf(star1[0], star2[0])
            var sy = minOf(star1[1], star2[1])
            var cnt = 0

            for (star in arr) {
                if (sx <= star[0] && star[0] <= sx + l &&
                    sy <= star[1] && star[1] <= sy + l
                ) {
                    cnt += 1
                }
            }
            count = maxOf(count, cnt)
        }
    }

    return (k - count).toString()
}

File: ddingmin/baekjoon/14658.py
import sys


def solve(n, m, l, k, arr):
    answer = 0
    for i in range(k):
        for j in range(k):
            cnt = 0

            # 두 별을 걸치는 가장 왼쪽 상단 좌표
            sx, sy = min(arr[i][0], arr[j][0]), min(arr[i][1], arr[j][1])

            for x, y in arr:
                # 구한 좌표의 최대 범위 별 개수 구하기
                if sx <= x <= sx + l and sy <= y <= sy + l:
                    cnt += 1
            answer = max(answer, cnt)

    return k - answer


def main():
    n, m, l, k = map(int, input().split())
    arr = []
    for _ in range(k):
        arr.append(list(map(int, input().split())))
    print(solve(n, m, l, k, arr))


if __name__ == '__main__':
    sys.setrecursionlimit(10 ** 5)
    input = sys.stdin.readline
    main()
